src/backend/columnar/columnar_tableam.c

```diff
@@ -2126,9 +2128,87 @@ ColumnarProcessUtility(PlannedStmt *pstmt,
 
 		RelationClose(rel);
 	}
+	else if (IsA(parsetree, CreateStmt))
+	{
+		CreateStmt *createStmt = castNode(CreateStmt, parsetree);
+		if (createStmt->accessMethod != NULL &&
+			!strcmp(createStmt->accessMethod, COLUMNAR_AM_NAME))
+		{
+			columnarRangeVar = createStmt->relation;
+			createStmt->options = ExtractColumnarRelOptions(createStmt->options,
+															&columnarOptions);
+		}
+	}
+	else if (IsA(parsetree, CreateTableAsStmt))
+	{
+		CreateTableAsStmt *createTableAsStmt = castNode(CreateTableAsStmt, parsetree);
+		IntoClause *into = createTableAsStmt->into;
+		if (into->accessMethod != NULL &&
+			!strcmp(into->accessMethod, COLUMNAR_AM_NAME))
+		{
+			columnarRangeVar = into->rel;
+			into->options = ExtractColumnarRelOptions(into->options,
+													  &columnarOptions);
+		}
+	}
+	else if (IsA(parsetree, AlterTableStmt))
+	{
+		AlterTableStmt *alterTableStmt = castNode(AlterTableStmt, parsetree);
+		bool toColumnar = false;
+
+		Relation rel = relation_openrv_extended(alterTableStmt->relation, AccessShareLock,
+												alterTableStmt->missing_ok);
+
+		if (rel != NULL){
+		bool isColumnar = rel->rd_tableam == GetColumnarTableAmRoutine();
+
+		ListCell *lc;
+		foreach (lc, alterTableStmt->cmds)
+		{
+			AlterTableCmd *alterTableCmd = castNode(AlterTableCmd, lfirst(lc));
+
+			if (alterTableCmd->subtype == AT_SetRelOptions ||
+				alterTableCmd->subtype == AT_ResetRelOptions)
+			{
+				List *options = castNode(List, alterTableCmd->def);
+
+				if (isColumnar || toColumnar)
+				{
+					columnarRangeVar = alterTableStmt->relation;
+					alterTableCmd->def = (Node *) ExtractColumnarRelOptions(options,
+																			&columnarOptions);
+				}
+			}
+#if PG_VERSION_NUM >= PG_VERSION_15
+			else if (alterTableCmd->subtype == AT_SetAccessMethod)
+			{
+				if (toColumnar || columnarRangeVar || columnarOptions)
+				{
+					ereport(ERROR, (errmsg("ALTER TABLE cannot alter the access method after altering storage parameters"),
+									errhint("Specify SET ACCESS METHOD before storage parameters, or use separate ALTER TABLE commands.")));
+				}
+
+				toColumnar = (strcmp(alterTableCmd->name, COLUMNAR_AM_NAME) == 0);
+
+				if (isColumnar && !toColumnar)
+				{
+					DeleteColumnarTableOptions(RelationGetRelid(rel), true);
+				}
+			}
+#endif /* PG_VERSION_15 */
+		}
+
+		relation_close(rel, NoLock);
+		}
+	}
 
 	PrevProcessUtilityHook_compat(pstmt, queryString, false, context,
 								  params, queryEnv, dest, completionTag);
+
+	if (columnarRangeVar != NULL && columnarOptions != NIL)
```

Comment:
list should generally not be compared to `NIL` iirc, instead `list_length() > 0`. Unless there is a semantic difference between no list, and a list of length 0.

Reply:
Hmm. That's all over the postgres code base, is this a citus coding standard thing?

Reply:
looking at the postgres source I find the following comment:
```
/*
 * The *only* valid representation of an empty list is NIL; in other
 * words, a non-NIL list is guaranteed to have length >= 1.
 */
#define NIL						((List *) NULL)
```

So I guess this is ok, probably messed up the notion of empty lists in golang.